src/Core/Services/CipherService.cs

```diff
@@ -571,28 +571,39 @@ public async Task SaveWithServerAsync(Cipher cipher)
             await UpsertAsync(data);
         }
 
-        public async Task ShareWithServerAsync(CipherView cipher, string organizationId, HashSet<string> collectionIds)
+        public async Task ShareWithServerAsync(CipherView cipherView, string organizationId, HashSet<string> collectionIds)
         {
             var attachmentTasks = new List<Task>();
-            if (cipher.Attachments != null)
+            Cipher cipher = null;
+            //If the cipher doesn't have a key, we update it 
+            if(await ShouldUseCipherKeyEncryptionAsync() && cipherView.Key == null)
```

Comment:
⛏️ Invert the order so it doesn't waste time firing the task when `Key != null`
```suggestion
            if (cipherView.Key == null && await ShouldUseCipherKeyEncryptionAsync())
```